lib/quantizeAttributes.js

```diff
@@ -0,0 +1,281 @@
+    var isQuantized = false;
+    for (var accessorId in accessorIds) {
+        if (accessorIds.hasOwnProperty(accessorId)) {
+            var accessor = accessors[accessorId];
+            var bufferViewId = accessor.bufferView;
+            var bufferView = bufferViews[bufferViewId];
+            var bufferId = bufferView.buffer;
+            var buffer = buffers[bufferId];
+            accessor.extensions = {
+                "WEB3D_quantized_attributes": {}
+            };
+            var attributes = accessor.extensions.WEB3D_quantized_attributes;
+            // accessor min and max are the extremes of the range
+            var min = accessor.min;
+            var max = accessor.max;
+            accessor.min = Array(min.length).fill(0);
+            accessor.max = Array(max.length).fill(range);
+            var decodeMatrix = createDecodeMatrix(min, max, range);
+
+            // change matrix precision to save space
+            if (defined(precision)) {
+                for (i = 0; i < decodeMatrix.length; i++) {
+                    var value = decodeMatrix[i];
+                    decodeMatrix[i] = Math.floor(value * precision) / precision;
+                }
+            }
+            attributes.decodedMin = min;
+            attributes.decodedMax = max;
+            attributes.decodeMatrix = decodeMatrix;
+
+            // Quantize the data in place
+            var source = buffer.extras._pipeline.source;
+            var offset = accessor.byteOffset + bufferView.byteOffset;
+            var num = 0;
+            var count = accessor.count;
+            var byteStride = getAccessorByteStride(accessor);
+            accessor.byteStride = byteStride;
+            var componentByteLength = byteLengthForComponentType(accessor.componentType);
+            var numberOfComponents = numberOfComponentsForType(accessor.type);
+            for (i = offset; num < count; i+=byteStride) {
+                for (var j = 0; j < numberOfComponents; j++) {
+                    var value = source.readFloatLE(i + j * componentByteLength);
+                    var encoded = Math.round((value - min[j]) * range / (max[j] - min[j]));
+                    source.writeUInt16LE(encoded, i + j * 2);
+                }
+                num++;
+            }
+            accessor.componentType = 5123;
+            isQuantized = true;
+        }
+    }
+    if (isQuantized) {
+        // Repack the buffers
+        removeUnusedVertices(gltf);
+        // Finalize
+        addExtensionsUsed(gltf, 'WEB3D_quantized_attributes');
+    }
+}
+
+function createDecodeMatrix(min, max, range) {
+    var size = min.length + 1;
+    if (size === 3) {
+        return createDecodeMatrix3(min, max, range);
+    } else if (size === 4) {
+        return createDecodeMatrix4(min, max, range);
+    } else if (size === 5) {
+        return createDecodeMatrix5(min, max, range);
+    }
+}
+
+function createDecodeMatrix3(min, max, range) {
```

Comment:
Here and below, the intention is for the whole numbers to be floating-point, right?  Append `.0`.

See https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Documentation/Contributors/CodingGuide/README.md#basic-code-construction